packages/blocks/src/__internal__/utils/selection.ts

```diff
@@ -49,43 +142,16 @@ function fixCurrentRangeToText(
       if (!text) {
         throw new Error('Failed to focus text node!');
       }
-      if (isForward) {
-        const rect = text.getBoundingClientRect();
-        const y = rect.bottom - 6;
-        newRange = caretRangeFromPoint(x, y);
-        if (newRange) {
-          if (!(newRange.endContainer.nodeType === Node.TEXT_NODE)) {
-            const lastTextNode = getLastTextNode(newRange.endContainer);
-            if (lastTextNode) {
-              newRange = document.createRange();
-              newRange.setStart(
-                lastTextNode,
-                lastTextNode.textContent?.length || 0
-              );
-              newRange.setEnd(
-                lastTextNode,
-                lastTextNode.textContent?.length || 0
-              );
-            }
-          }
-          range.setEnd(newRange.endContainer, newRange.endOffset);
-        }
-      } else {
-        const rect = text.getBoundingClientRect();
-        const y = rect.top + 6;
-        newRange = caretRangeFromPoint(x, y);
-        if (newRange) {
-          if (!(newRange.startContainer.nodeType === Node.TEXT_NODE)) {
-            const firstTextNode = getFirstTextNode(newRange.startContainer);
-            if (firstTextNode) {
-              newRange = document.createRange();
-              newRange.setStart(firstTextNode, 0);
-              newRange.setEnd(firstTextNode, 0);
-            }
-          }
-          range.setStart(newRange.endContainer, newRange.endOffset);
-        }
-      }
+      const rect = text.getBoundingClientRect();
+      const NewY = isForward ? rect.bottom - 6 : rect.top + 6;
+      newRange = caretRangeFromPoint(x, NewY);
+      isForward
```

Comment:
What is the `null` used for here? Plain if-else looks fine here. Is this what you want to express?

``` ts
if (isForward && newRange) {
  forwardSelect(newRange, range);
}
else if (!isForward && newRange) {
  backwardSelect(newRange, range);
}
```